6_wqp_fetch/src/fetch_wqp_data.R

```diff
@@ -136,21 +134,24 @@ pull_wqp_data <- function(data_file, wqp_sites, characteristicName, dummy){
 
   charnames <- str_split(characteristicName, '\\|')[[1]]
 
-
   # prepare the arguments to pass to readWQPdata
   wqp_args <- list()
   wqp_args$characteristicName <- charnames
-  if (is.na(wqp_sites$MonitoringLocationIdentifier[1])){
-    wqp_args$siteid <- pull(wqp_sites, MonitoringLocationIdentifier)
-  } else {
+
+  if (wqp_sites$pull_org[1]){
     wqp_args$organization <- pull(wqp_sites, OrganizationIdentifier)
+  } else {
+    wqp_args$siteid <- pull(wqp_sites, MonitoringLocationIdentifier)
   }
 
 
   # do the data pull
   # first pull using readWQPdata, then if that fails, try POST
   wqp_dat <- suppressMessages(wqp_POST(wqp_args))
 
+  if (wqp_sites$pull_org){
```

Comment:
Do you need a subset here, as above (`wqp_sites$pull_org[1]`)?

Reply:
yes, this one needs that (same as above), but works like this just with an warning that it is using the first of the vector. Will update this one too.